pkgs/applications/science/biology/blast/default.nix

```diff
@@ -0,0 +1,98 @@
+{ stdenv, fetchurl, zlib, bzip2, boost, perl, cpio, gawk, coreutils }:
+
+stdenv.mkDerivation rec {
+  pname = "blast";
+  version = "2.9.0";
+
+  src = fetchurl {
+    url = "ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/${version}/ncbi-blast-${version}+-src.tar.gz";
+    sha256 = "0my2rpd1bln05sxnp4c5wk5j5y6yx56vi38pzicjfhh9g8nwr453";
+  };
+
+  sourceRoot = "ncbi-blast-${version}+-src/c++";
+  
+  configureFlags = [ "--without-makefile-auto-update" ];
+  preConfigure = ''
+    export NCBICXX_RECONF_POLICY=warn
+    export PWD=$(pwd)
+    export HOME=$PWD
+
+    # The configure scripts wants to set AR="ar cr" unless it is already set in
+    # the environment. Because stdenv sets AR="ar", the result is a bad call to
+    # the assembler later in the process. Thus, we need to unset AR
+    unset AR
+
+    for awks in scripts/common/impl/is_log_interesting.awk \
+        scripts/common/impl/report_duplicates.awk; do
+
+        substituteInPlace $awks \
+              --replace /usr/bin/awk ${gawk}/bin/awk
+    done
+
+    for mk in src/build-system/Makefile.meta.in \
+        src/build-system/helpers/run_with_lock.c ; do
+
+        substituteInPlace $mk \
+        --replace /bin/rm ${coreutils}/bin/rm
+    done
+
+    for mk in src/build-system/Makefile.meta.gmake=no \
+        src/build-system/Makefile.meta_l \
+        src/build-system/Makefile.meta_r \
+        src/build-system/Makefile.requirements \
+        src/build-system/Makefile.rules_with_autodep.in; do
+
+        substituteInPlace $mk \
+            --replace /bin/echo ${coreutils}/bin/echo
+    done
+    for mk in src/build-system/Makefile.meta_p \
+        src/build-system/Makefile.rules_with_autodep.in; do
+
+        substituteInPlace $mk \
+            --replace /bin/mv ${coreutils}/bin/mv
+    done
+
+
+    substituteInPlace src/build-system/configure \
+        --replace /bin/pwd ${coreutils}/bin/pwd \
+        --replace /bin/ln ${coreutils}/bin/ln
+
+    substituteInPlace src/build-system/configure.ac \
+        --replace /bin/pwd ${coreutils}/bin/pwd \
+        --replace /bin/ln ${coreutils}/bin/ln
+
+    substituteInPlace src/build-system/Makefile.meta_l \
+        --replace /bin/date ${coreutils}/bin/date
+  '';
+
+  nativeBuildInputs = [ perl ];
+
+  buildInputs = [ coreutils gawk zlib boost bzip2 cpio ];
+  hardeningDisable = [ "format" ];
+
+  patches = [ ./no_slash_bin.patch ];
```

Comment:
Isn't this going to be a pain to maintain? The whole patch may need to be reworked for each new version

Reply:
Yep. I can replace one or two instances of patching with `substituteInPlace`, but otherwise, this is a pretty hostile build script: it repeatedly tries to call utilities in `/bin/`

Reply:
I replaced the uses of the patch that I could with `substituteInPlace`, but the remaining ones don't seem to lend themselves to doing so.

Reply:
Few days before I noticed this PR I was trying to get blast compiling in my `nur-packages`, I used a very coarse-grained patching https://github.com/pschuprikov/nur-packages/blob/d15609f3442ed4dac23d9d782f78cd734221e5b4/pkgs/bioinf/ncbi_blast/generic.nix#L19-L38. 

@luispedro if you feel that could be a better way to go, I can create a PR against your branch to save your time.